app/knights/heroes.py

```diff
@@ -0,0 +1,36 @@
+class Hero:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            protection: int = 0
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = protection
+
+
+def init_hero(knights: dict) -> list[Hero]:
```

Comment:
Use plural form since you return a list of elements
```suggestion
def init_heros(knights: dict) -> list[Hero]:
```